src/main/java/org/wildfly/security/auth/server/ServerAuthenticationContext.java

```diff
@@ -1031,6 +1034,7 @@ private void handleOne(final Callback[] callbacks, final int idx) throws IOExcep
                         log.tracef("Handling CachedIdentityAuthorizeCallback: principal = %s  authorizedIdentity = %s", principal, authorizedIdentity);
```

Comment:
@dmlloyd I've removed the try/finally block and return statements. In case an exception is thrown we should not call ```handleOne```.